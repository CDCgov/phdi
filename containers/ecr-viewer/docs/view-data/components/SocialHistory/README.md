[**ecr-viewer**](../../../README.md) â€¢ **Docs**

***

[ecr-viewer](../../../README.md) / view-data/components/SocialHistory

# view-data/components/SocialHistory

## Index

### Functions

- [default](functions/default.md)
