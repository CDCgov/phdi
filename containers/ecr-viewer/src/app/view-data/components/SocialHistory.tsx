import { DisplayData } from "@/app/utils";

interface SocialHistoryProps {
    socialData: DisplayData[]
}

const SocialHistory = (
    { socialData }: SocialHistoryProps
) => {
    const renderDemographicsData = (item: any, index: number) => {
        return (
            <div key={index}>
                <div className="grid-row">
                    <div className="data-title"><h4>{item.title}</h4></div>
                    <div className="grid-col-auto maxw7">
                        {item.value}
                    </div>
                </div>
                <div className={"section__line_gray"} />
            </div>
        )
    }

    return (
        <div>
            <div
                className="padding-bottom-3"
                aria-labelledby="summary-box-key-information"
            >
                <div className="usa-summary-box__body">
                    <h3
                        className="usa-summary-box__heading padding-y-105"
                        id="summary-box-key-information"
                    >
                        Social History
                    </h3>
                    <div className="usa-summary-box__text">
                        {socialData.map((item, index) => renderDemographicsData(item, index))}
                    </div>
                </div>
            </div>
        </div>);
};

export default SocialHistory;