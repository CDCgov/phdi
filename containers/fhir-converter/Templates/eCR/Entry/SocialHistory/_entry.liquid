{% if entry.observation -%}
    {% if entry.observation.code.nullFlavor or entry.observation.value.nullFlavor -%}
    {% else -%}
        {% assign observationId = entry.observation | to_json_string | generate_uuid -%}
        {% include 'Resource/Observation' observationCategory: 'social-history', observationEntry: entry.observation, ID: observationId -%}
        {% include 'Reference/Observation/Subject' ID: observationId, REF: fullPatientId -%}
    {% endif -%}
{% endif -%}
