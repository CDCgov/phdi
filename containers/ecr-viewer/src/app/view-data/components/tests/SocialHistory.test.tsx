import { render } from "@testing-library/react";
import { axe } from "jest-axe";
import SocialHistory from "../SocialHistory";

describe("SocialHistory", () => {
  let container: HTMLElement;
  beforeAll(() => {
    const socialAvailableData = [
      {
        title: "Occupation",
        value: "Nursing, psychiatric, and home health aides",
      },
      {
        title: "Tobacco Use",
        value: "Tobacco smoking consumption unknown",
      },
      {
        title: "Homeless Status",
        value: "unsatisfactory living conditions (finding)",
      },
      {
        title: "Occupation",
        value: "Nursing, psychiatric, and home health aides",
      },
    ];
    container = render(
      <SocialHistory socialData={socialAvailableData} />,
    ).container;
  });
  it("should match snapshot", () => {
    expect(container).toMatchSnapshot();
  });
  it("should pass accessibility test", async () => {
    expect(await axe(container)).toHaveNoViolations();
  });
});
