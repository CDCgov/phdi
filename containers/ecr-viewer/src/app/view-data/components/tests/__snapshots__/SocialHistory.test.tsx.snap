// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SocialHistory should match snapshot 1`] = `
<div>
  <div>
    <div
      class="padding-bottom-3"
    >
      <div
        class="usa-summary-box__body"
      >
        <h4
          class="usa-summary-box__heading padding-y-105"
          id="summary-box-key-information"
        >
          <span
            id="social-history"
          >
            Social History
          </span>
        </h4>
        <div
          class="usa-summary-box__text"
        >
          <div>
            <div
              class="grid-row"
            >
              <div
                class="data-title"
              >
                Occupation
              </div>
              <div
                class="grid-col-auto maxw7 text-pre-line"
              >
                Nursing, psychiatric, and home health aides
              </div>
            </div>
            <div
              class="section__line_gray"
            />
          </div>
          <div>
            <div
              class="grid-row"
            >
              <div
                class="data-title"
              >
                Tobacco Use
              </div>
              <div
                class="grid-col-auto maxw7 text-pre-line"
              >
                Tobacco smoking consumption unknown
              </div>
            </div>
            <div
              class="section__line_gray"
            />
          </div>
          <div>
            <div
              class="grid-row"
            >
              <div
                class="data-title"
              >
                Homeless Status
              </div>
              <div
                class="grid-col-auto maxw7 text-pre-line"
              >
                unsatisfactory living conditions (finding)
              </div>
            </div>
            <div
              class="section__line_gray"
            />
          </div>
          <div>
            <div
              class="grid-row"
            >
              <div
                class="data-title"
              >
                Occupation
              </div>
              <div
                class="grid-col-auto maxw7 text-pre-line"
              >
                Nursing, psychiatric, and home health aides
              </div>
            </div>
            <div
              class="section__line_gray"
            />
          </div>
          <div>
            <div
              class="grid-row"
            >
              <div
                class="data-title"
              >
                Sexual Orientation
              </div>
              <div
                class="grid-col-auto maxw7 text-pre-line"
              >
                Do not know
              </div>
            </div>
            <div
              class="section__line_gray"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
